qiskit_ibm_runtime/session.py

```diff
@@ -138,9 +140,30 @@ def run(
 
         Returns:
             Submitted job.
+
+        Raises:
+            IBMInputValueError: If a backend is passed in through options that does not match
+                the current session backend.
         """
 
         options = options or {}
+        if "backend" in options:
+            issue_deprecation_msg(
+                "'backend' is no longer a supported option within a session",
+                "0.8",
+                "Sessions do not support multiple backends.",
+                3,
+            )
+            if (
+                self._session_id
```

Comment:
I don't think we need to compare session id here? If I do

```
session = Session(backend="foo")
session.run(options={"backend": "bar"})
```
^ this should fail because it's ambiguous which backend I really want.